aws/tests/test_deregister_amis.py

```diff
@@ -0,0 +1,122 @@
+"""
+Tests for testeng-ci/aws.
+"""
+import boto
+import logging
+from aws.deregister_amis import main
+from mock import patch
+from moto import mock_ec2
+from testfixtures import LogCapture
+from unittest import TestCase
+
+
+class DeregisterAmisTestCase(TestCase):
+    """
+    TestCase class for testing get_running_instances.py.
+    """
+
+    def setUp(self):
+        self.key_id = 'my-key-id'
+        self.secret_key = 'my-secret-key'
+        self.conn = boto.connect_ec2(self.key_id, self.secret_key)
+
+    def _get_test_image(self):
+        test_image_id = 'ami-11122278'
+        reservation = self.conn.run_instances(test_image_id)
+        self.conn.create_image(
+            name='test-ami',
+            instance_id=reservation.instances[0].id
+        )
+        return self.conn.get_all_images()[0]
+
+    @mock_ec2
+    def test_main(self):
+        """
+        Test output of main
+        """
+        args = [
+            '-i', self.key_id,
+            '-s', self.secret_key,
+            '--log-level', 'INFO',
+        ]
+
+        with LogCapture() as l:
+            main(args)
+            output = (
+                'aws.deregister_amis',
+                'INFO',
+                'Finding AMIs tagged with delete_or_keep: delete'
```

Comment:
Honestly that does not work. I tried several times and in several ways, and the way the tuple is understood by python seems to change whether this is extracted or not. It's weird. I can show you if you're curious.